src/array/coord/interleaved/builder.rs

```diff
@@ -71,10 +71,18 @@ impl<const D: usize> InterleavedCoordBufferBuilder<D> {
         self.coords.extend_from_slice(&c);
     }
 
-    pub fn push_point(&mut self, point: &impl PointTrait<T = f64>) {
-        let buf: [f64; D] = core::array::from_fn(|i| point.nth(i).unwrap_or(f64::NAN));
+    pub fn push_coord(&mut self, coord: &impl CoordTrait<T = f64>) {
+        let buf: [f64; D] = core::array::from_fn(|i| coord.nth(i).unwrap_or(f64::NAN));
         self.coords.extend_from_slice(&buf);
     }
+
+    pub fn push_point(&mut self, point: &impl PointTrait<T = f64>) {
+        if let Some(coord) = point.coord() {
+            self.push_coord(&coord);
+        } else {
+            self.push([f64::NAN; D]);
```

Comment:
I'm completely out of my depth with geoarrow, but this appears a little awkward. Does the `push_point` method even make sense anymore?

Reply:
It's a little awkward, and potentially could be refactored in the future/moved to the point builder, but at some level there needs to be a conversion between empty points and the GeoArrow representation of those, which is with NAN values.

Reply:
Yeah, I guess this comes down to my unfamiliarity with geoarrow. 

So it sounds like a `PointArray` needs to have an element in its `CoordBuffer` for each point, even if that point is `POINT EMPTY`. 

Is the `validity: NullBuffer` used to encode `POINT EMPTY`?

Reply:
> So it sounds like a `PointArray` needs to have an element in its `CoordBuffer` for each point, even if that point is `POINT EMPTY`.

Correct. 

The raw layout for a 2D point array with 4 points is:

```py
[0, 2, 4, 5, NaN, NaN, 10, 2]
```

This translates to 
```
Point(0, 2),
Point(4, 5),
Point EMPTY,
Point(10, 2)
```

> Is the `validity: NullBuffer` used to encode which coords correspond to `POINT EMPTY`?

_No_... With GeoArrow we have **three** states of geometry:

- `null`: this is a row that does not have a geometry
- `empty`: this row has a geometry but its geometry is empty (e.g. as a result of an empty intersection)
- valid and non-empty: a standard geometry

The `validity: NullBuffer` encodes the first of those. Every Arrow array has an optional nullability buffer, and that can say "this row is null".

When that validity buffer says that a row is valid, then we can look into the encoded geometry buffer, but it's still possible for that encoded geometry to be empty.

So for a 2D point, the encoding is one of these three states:

- `null`: validity buffer indicates value is null. For a point array the memory still needs to be _reserved_ so that the offsets of future points are correct, but the values of that point's coordinates are undefined. 
- `empty`: validity buffer indicates not null. But all values are `f64::NAN`. 
- `valid`: validity buffere not null, values are not `f64::NAN`.